### ISSUE TITLE
NOAUTH Authentication required

### ISSUE THREAD

**Issue by Dzordzu** [netbox-community/netbox-docker#140]
### Description
After solving the #139 I've found other issue. REDIS seems to reject given authorization details

```
<class 'redis.exceptions.ResponseError'>

NOAUTH Authentication required.
```

### System info
Kernel 5.1.12
Docker 18.09.6-ce
Compose 1.24.0

#### Containers
Redis 4.0.14

### Logs
Logs seem to be fine

#### `docker-compose logs -f netbox-worker`
No output

#### `docker-compose logs -f netbox`
```
netbox_1         | Operations to perform:
netbox_1         |   Apply all migrations: admin, auth, circuits, contenttypes, dcim, extras, ipam, secrets, sessions, taggit, tenancy, users, virtualization
netbox_1         | Running migrations:
netbox_1         |   No migrations to apply.
netbox_1         | 💡 Username: admin, E-Mail: admin@example.com
netbox_1         | ⚙️ Executing '/opt/netbox/startup_scripts/000_users.py'
netbox_1         | ⚙️ Executing '/opt/netbox/startup_scripts/010_groups.py'
netbox_1         | ⚙️ Executing '/opt/netbox/startup_scripts/020_custom_fields.py'
netbox_1         | ⚙️ Executing '/opt/netbox/startup_scripts/030_regions.py'
netbox_1         | ⚙️ Executing '/opt/netbox/startup_scripts/040_sites.py'
netbox_1         | ⚙️ Executing '/opt/netbox/startup_scripts/050_manufacturers.py'
netbox_1         | ⚙️ Executing '/opt/netbox/startup_scripts/060_device_types.py'
netbox_1         | ⚙️ Executing '/opt/netbox/startup_scripts/070_rack_roles.py'
netbox_1         | ⚙️ Executing '/opt/netbox/startup_scripts/080_racks.py'
netbox_1         | ⚙️ Executing '/opt/netbox/startup_scripts/090_device_roles.py'
netbox_1         | ⚙️ Executing '/opt/netbox/startup_scripts/100_platforms.py'
netbox_1         | ⚙️ Executing '/opt/netbox/startup_scripts/110_devices.py'
netbox_1         | 
netbox_1         | 0 static files copied to '/opt/netbox/netbox/static', 344 unmodified.
netbox_1         | ✅ Initialisation is done.
netbox_1         | [2019-06-21 12:47:17 +0000] [1] [DEBUG] Current configuration:
netbox_1         |   config: /etc/netbox/config/gunicorn_config.py
netbox_1         |   bind: ['0.0.0.0:8001']
netbox_1         |   backlog: 2048
netbox_1         |   workers: 3
netbox_1         |   worker_class: sync
netbox_1         |   threads: 1
netbox_1         |   worker_connections: 1000
netbox_1         |   max_requests: 0
netbox_1         |   max_requests_jitter: 0
netbox_1         |   timeout: 30
netbox_1         |   graceful_timeout: 30
netbox_1         |   keepalive: 2
netbox_1         |   limit_request_line: 4094
netbox_1         |   limit_request_fields: 100
netbox_1         |   limit_request_field_size: 8190
netbox_1         |   reload: False
netbox_1         |   reload_engine: auto
netbox_1         |   reload_extra_files: []
netbox_1         |   spew: False
netbox_1         |   check_config: False
netbox_1         |   preload_app: False
netbox_1         |   sendfile: None
netbox_1         |   reuse_port: False
netbox_1         |   chdir: /opt/netbox/netbox
netbox_1         |   daemon: False
netbox_1         |   raw_env: []
netbox_1         |   pidfile: None
netbox_1         |   worker_tmp_dir: None
netbox_1         |   user: 0
netbox_1         |   group: 0
netbox_1         |   umask: 0
netbox_1         |   initgroups: False
netbox_1         |   tmp_upload_dir: None
netbox_1         |   secure_scheme_headers: {'X-FORWARDED-PROTOCOL': 'ssl', 'X-FORWARDED-PROTO': 'https', 'X-FORWARDED-SSL': 'on'}
netbox_1         |   forwarded_allow_ips: ['127.0.0.1']
netbox_1         |   accesslog: -
netbox_1         |   disable_redirect_access_to_syslog: False
netbox_1         |   access_log_format: %(h)s %(l)s %(u)s %(t)s "%(r)s" %(s)s %(b)s "%(f)s" "%(a)s"
netbox_1         |   errorlog: -
netbox_1         |   loglevel: debug
netbox_1         |   capture_output: False
netbox_1         |   logger_class: gunicorn.glogging.Logger
netbox_1         |   logconfig: None
netbox_1         |   logconfig_dict: {}
netbox_1         |   syslog_addr: udp://localhost:514
netbox_1         |   syslog: False
netbox_1         |   syslog_prefix: None
netbox_1         |   syslog_facility: user
netbox_1         |   enable_stdio_inheritance: False
netbox_1         |   statsd_host: None
netbox_1         |   statsd_prefix: 
netbox_1         |   proc_name: None
netbox_1         |   default_proc_name: netbox.wsgi
netbox_1         |   pythonpath: /opt/netbox/netbox
netbox_1         |   paste: None
netbox_1         |   on_starting: <function OnStarting.on_starting at 0x7fed4df4c158>
netbox_1         |   on_reload: <function OnReload.on_reload at 0x7fed4df4c268>
netbox_1         |   when_ready: <function WhenReady.when_ready at 0x7fed4df4c378>
netbox_1         |   pre_fork: <function Prefork.pre_fork at 0x7fed4df4c488>
netbox_1         |   post_fork: <function Postfork.post_fork at 0x7fed4df4c598>
netbox_1         |   post_worker_init: <function PostWorkerInit.post_worker_init at 0x7fed4df4c6a8>
netbox_1         |   worker_int: <function WorkerInt.worker_int at 0x7fed4df4c7b8>
netbox_1         |   worker_abort: <function WorkerAbort.worker_abort at 0x7fed4df4c8c8>
netbox_1         |   pre_exec: <function PreExec.pre_exec at 0x7fed4df4c9d8>
netbox_1         |   pre_request: <function PreRequest.pre_request at 0x7fed4df4cae8>
netbox_1         |   post_request: <function PostRequest.post_request at 0x7fed4df4cb70>
netbox_1         |   child_exit: <function ChildExit.child_exit at 0x7fed4df4cc80>
netbox_1         |   worker_exit: <function WorkerExit.worker_exit at 0x7fed4df4cd90>
netbox_1         |   nworkers_changed: <function NumWorkersChanged.nworkers_changed at 0x7fed4df4cea0>
netbox_1         |   on_exit: <function OnExit.on_exit at 0x7fed4df57048>
netbox_1         |   proxy_protocol: False
netbox_1         |   proxy_allow_ips: ['127.0.0.1']
netbox_1         |   keyfile: None
netbox_1         |   certfile: None
netbox_1         |   ssl_version: 2
netbox_1         |   cert_reqs: 0
netbox_1         |   ca_certs: None
netbox_1         |   suppress_ragged_eofs: True
netbox_1         |   do_handshake_on_connect: False
netbox_1         |   ciphers: TLSv1
netbox_1         |   raw_paste_global_conf: []
netbox_1         | [2019-06-21 12:47:17 +0000] [1] [INFO] Starting gunicorn 19.9.0
netbox_1         | [2019-06-21 12:47:17 +0000] [1] [DEBUG] Arbiter booted
netbox_1         | [2019-06-21 12:47:17 +0000] [1] [INFO] Listening at: http://0.0.0.0:8001 (1)
netbox_1         | [2019-06-21 12:47:17 +0000] [1] [INFO] Using worker: sync
netbox_1         | [2019-06-21 12:47:17 +0000] [187] [INFO] Booting worker with pid: 187
netbox_1         | [2019-06-21 12:47:17 +0000] [193] [INFO] Booting worker with pid: 193
netbox_1         | [2019-06-21 12:47:17 +0000] [194] [INFO] Booting worker with pid: 194
netbox_1         | [2019-06-21 12:47:17 +0000] [1] [DEBUG] 3 workers
netbox_1         | [2019-06-21 12:47:20 +0000] [193] [DEBUG] GET /
netbox_1         | 172.21.0.5 - - [21/Jun/2019:12:47:20 +0000] "GET / HTTP/1.0" 500 1759 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:67.0) Gecko/20100101 Firefox/67.0"
```

#### `docker-compose logs -f redis`
```
redis_1          | 1:C 21 Jun 12:05:23.215 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
redis_1          | 1:C 21 Jun 12:05:23.215 # Redis version=4.0.14, bits=64, commit=00000000, modified=0, pid=1, just started
redis_1          | 1:C 21 Jun 12:05:23.215 # Configuration loaded
redis_1          | 1:M 21 Jun 12:05:23.217 * Running mode=standalone, port=6379.
redis_1          | 1:M 21 Jun 12:05:23.217 # WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.
redis_1          | 1:M 21 Jun 12:05:23.217 # Server initialized
redis_1          | 1:M 21 Jun 12:05:23.217 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
redis_1          | 1:M 21 Jun 12:05:23.217 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with Redis. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. Redis must be restarted after THP is disabled.
redis_1          | 1:M 21 Jun 12:05:23.217 * DB loaded from append only file: 0.000 seconds
redis_1          | 1:M 21 Jun 12:05:23.217 * Ready to accept connections
```

#### `docker-compose logs -f postgres`
```
postgres_1       | 2019-06-21 10:08:31.572 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgres_1       | 2019-06-21 10:08:31.572 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgres_1       | 2019-06-21 10:08:31.578 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgres_1       | 2019-06-21 10:08:31.594 UTC [18] LOG:  database system was shut down at 2019-06-21 10:08:18 UTC
postgres_1       | 2019-06-21 10:08:31.597 UTC [1] LOG:  database system is ready to accept connections
postgres_1       | 2019-06-21 11:45:54.067 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgres_1       | 2019-06-21 11:45:54.068 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgres_1       | 2019-06-21 11:45:54.072 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgres_1       | 2019-06-21 11:45:54.084 UTC [18] LOG:  database system was interrupted; last known up at 2019-06-21 11:19:58 UTC
postgres_1       | 2019-06-21 11:45:54.533 UTC [18] LOG:  database system was not properly shut down; automatic recovery in progress
postgres_1       | 2019-06-21 11:45:54.536 UTC [18] LOG:  redo starts at 0/26F2570
postgres_1       | 2019-06-21 11:45:54.536 UTC [18] LOG:  invalid record length at 0/26F2650: wanted 24, got 0
postgres_1       | 2019-06-21 11:45:54.536 UTC [18] LOG:  redo done at 0/26F2618
postgres_1       | 2019-06-21 11:45:54.552 UTC [1] LOG:  database system is ready to accept connections
postgres_1       | 2019-06-21 11:58:02.108 UTC [1] LOG:  received smart shutdown request
postgres_1       | 2019-06-21 11:58:02.111 UTC [1] LOG:  worker process: logical replication launcher (PID 24) exited with exit code 1
postgres_1       | 2019-06-21 11:58:02.111 UTC [19] LOG:  shutting down
postgres_1       | 2019-06-21 11:58:02.143 UTC [1] LOG:  database system is shut down
postgres_1       | 2019-06-21 11:58:06.529 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgres_1       | 2019-06-21 11:58:06.530 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgres_1       | 2019-06-21 11:58:06.533 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgres_1       | 2019-06-21 11:58:06.546 UTC [18] LOG:  database system was shut down at 2019-06-21 11:58:02 UTC
postgres_1       | 2019-06-21 11:58:06.549 UTC [1] LOG:  database system is ready to accept connections

```

### Additional info
I tried to enable redis no auth required via docker-compose env, but it seems to be broken (it leads to other complications)

There were other attempts, but (to be honest) i don't remember what I've done

**Comment by Snoo-py** [comment]
Hi,

I had the same problem, I feel it's due to this commit in netbox https://github.com/digitalocean/netbox/commit/c2d0e8fd9557252e348fe938400de4c5c86f079d.

I fix temporarily this issue by adding in ~/netbox-docker/configuration/configuration.py
```
REDIS = {  
....  
    'CACHE_DATABASE': 0,  
....  
}   
CACHE_TIMEOUT = 0   
```

**Comment by cimnine** [comment]
There is a bug in Netbox, see https://github.com/digitalocean/netbox/issues/3282.

**Comment by cimnine** [comment]
Thank you for reporting this. This relates to a bug in Netbox v2.6.0. I will therefore close this issue.

**Comment by cimnine** [comment]
The workaround described above is discouraged (as described in [the Netbox `v2.6.0` release notes](https://github.com/digitalocean/netbox/releases/tag/v2.6.0)).

**tl;dr: You might loose pending webhook notifications!**

**Comment by arionl** [comment]
Sorry to comment on a closed issue - is there any better workaround than described above? At least until the upstream issue is addressed? (as I understand it, upstream will eventually add multiple config stanzas for different Redis endpoints). I really want to move to 2.6.x but I'm hesitant to do so without proper Redis config/caching..

**Comment by cimnine** [comment]
@arionl you should be able to use `2.6.1` as I've linked the wrong bug report above. The issue described by Snoo-py should be resolved in `2.6.1`.

Make sure you create a proper backup before upgrading so that you could go back eventually.

**Comment by arionl** [comment]
> @arionl you should be able to use `2.6.1` as I've linked the wrong bug report above. The issue described by Snoo-py should be resolved in `2.6.1`.
> 
> Make sure you create a proper backup before upgrading so that you could go back eventually.

Ahh, thank you! I had tried 2.6.0 and saw the NO_AUTH error but after pulling the 2.6.1 image I'm up and running on my test environment using your out-of-the-box config settings for Redis. Appreciate the call out!
